community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/ExpressionAcceptanceTest.scala

```diff
@@ -91,4 +91,66 @@ class ExpressionAcceptanceTest extends ExecutionEngineFunSuite with QueryStatist
       executeScalarWithAllPlannersAndCompatibilityMode[Any]("RETURN {expr}[{idx}]", "expr" -> 1, "idx" -> 12.3)
     }
   }
+
+  test("should handle map projection with property selectors") {
+    createNode("foo" -> 1, "bar" -> "apa")
+
+    executeScalarWithAllPlanners[Any]("MATCH (n) RETURN n{.foo,.bar,.baz}") should equal(
+      Map("foo" -> 1, "bar" -> "apa", "baz" -> null))
+  }
+
+  test("should handle map projection with property selectors and identifier selector") {
+    createNode("foo" -> 1, "bar" -> "apa")
+
+    executeScalarWithAllPlanners[Any]("WITH 42 as x MATCH (n) RETURN n{.foo,.bar,x}") should equal(
```

Comment:
That would return a map with the `foo` and `bar` values, and a separate column with the `x` value. This query is trying to include the values of the `x` variable in the map, not separately.